File: Python/Sorting/insertion-sort.py
def insertionSort(nums):
    for i in range(1, len(nums)):
        key = nums[i]
        j = i-1
        while j >= 0 and key < nums[j] :
                nums[j + 1] = nums[j]
                j -= 1
        nums[j + 1] = key
    return nums

nums = []
n = int(input("Enter n: "))
for i in range(n):
    element = int(input("Enter element {}: ".format(i+1)))
    nums.append(element)

print()
print("Current array:", nums)
insertionSort(nums)
print("Sorted array:", nums)

File: C/Sorting/insertion-sort.c
#include <stdio.h>

void insertionSort(int arr[], int n);
void display(int arr[], int n);

int main() {

    int arr[] = {4, 7, 1, 2, 8, 99};
    insertionSort(arr, 6);

}

void insertionSort(int arr[], int n) {

    int i, j, key;

    for (int i = 1; i < n; ++i) {
        key = arr[i];
        j = i-1;
        while (j >= 0 && arr[j] > key) {
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = key;
        display(arr, n);
    }

}

void display(int arr[], int n) {

    printf("Current array: ");
    for (int i = 0; i < n; ++i) {
        printf("%d ", arr[i]);
    }
    printf("\n");

}